### DISCUSSION TITLE
KS8 PODS - ADONIS JS

### DISCUSSION THREAD

**Discussion by Joaoof** [adonisjs/core#4221] [Help]
this error appears in my pod log: SyntaxError:
Cannot
use
import
statement
outside
a
module
 
   at (syntax error) /home/node/app/config/app.ts:8
    3|    *
    4|    * Feel free to let us know via PR, if you find something broken in this config
    5|    * file.
    6|    */
    7|   
 ❯  8|   import proxyAddr from 'proxy-addr'
    9|   import Env from '@ioc:Adonis/Core/Env'
   10|  import type { ServerConfig } from '@ioc:Adonis/Core/Server'
   11|  import type { LoggerConfig } from '@ioc:Adonis/Core/Logger'
   12|  import type { ProfilerConfig } from '@ioc:Adonis/Core/Profiler'
   13|  // import type { ValidatorConfig } from '@ioc:Adonis/Core/Validator'
   14|  
   15|  /*
   16|  |--------------------------------------------------------------------------
   17|  | Application secret key
   18|  |--------------------------------------------------------------------------
   19|  |
   20|  | The secret to encrypt and sign different values in your application.
   21|  | Make sure to keep the `APP_KEY` as an environment variable and secure.
   22|  |
   23|  | Note: Changing the application key for an existing app will make all
   24|  | the cookies invalid and also the existing encrypted data will not
   25|  | be decrypted.
   26|  |
   27|  */
   28|  export const appKey: string = Env.get('APP_KEY')
   29|  
   30|  /*
   31|  |--------------------------------------------------------------------------
   32|  | Http server configuration
   33|  |--------------------------------------------------------------------------
   34|  |
   35|  | The configuration for the HTTP(s) server. Make sure to go through all
   36|  | the config properties to make keep server secure.
   37|  |
   38|  */
   39|  export const http: ServerConfig = {
   40|    /*
   41|    |--------------------------------------------------------------------------
   42|    | Allow method spoofing
   43|    |--------------------------------------------------------------------------
   44|    |
   45|    | Method spoofing enables defining custom HTTP methods using a query string
   46|    | `_method`. This is usually required when you are making traditional
   47|    | form requests and wants to use HTTP verbs like `PUT`, `DELETE` and
   48|    | so on.
   49|    |
   50|    */
   51|    allowMethodSpoofing: false,
   52|  
   53|    /*
   54|    |--------------------------------------------------------------------------
   55|    | Subdomain offset
   56|    |--------------------------------------------------------------------------
   57|    */
   58|    subdomainOffset: 2,
   59|  
   60|    /*
   61|    |--------------------------------------------------------------------------
   62|    | Request Ids
   63|    |--------------------------------------------------------------------------
   64|    |
   65|    | Setting this value to `true` will generate a unique request id for each
   66|    | HTTP request and set it as `x-request-id` header.
   67|    |
   68|    */
   69|    generateRequestId: false,
   70|  
   71|    /*
   72|    |--------------------------------------------------------------------------
   73|    | Trusting proxy servers
   74|    |--------------------------------------------------------------------------
   75|    |
   76|    | Define the proxy servers that AdonisJs must trust for reading `X-Forwarded`
   77|    | headers.
   78|    |
   79|    */
   80|    trustProxy: proxyAddr.compile('loopback'),
   81|  
   82|    /*
   83|    |--------------------------------------------------------------------------
   84|    | Generating Etag
   85|    |--------------------------------------------------------------------------
   ⁃ anonymous
     /home/node/app/node_modules/require-all/index.js:56. how to solve this adonisjs syntax problem? my app.ts: /**
 * Config source: https://git.io/JfefZ
 *
 * Feel free to let us know via PR, if you find something broken in this config
 * file.
 */

import proxyAddr from 'proxy-addr'
import Env from '@ioc:Adonis/Core/Env'
import type { ServerConfig } from '@ioc:Adonis/Core/Server'
import type { LoggerConfig } from '@ioc:Adonis/Core/Logger'
import type { ProfilerConfig } from '@ioc:Adonis/Core/Profiler'
import type { ValidatorConfig } from '@ioc:Adonis/Core/Validator'

/*
|--------------------------------------------------------------------------
| Application secret key
|--------------------------------------------------------------------------
|
| The secret to encrypt and sign different values in your application.
| Make sure to keep the `APP_KEY` as an environment variable and secure.
|
| Note: Changing the application key for an existing app will make all
| the cookies invalid and also the existing encrypted data will not
| be decrypted.
|
*/
export const appKey: string = Env.get('APP_KEY')

/*
|--------------------------------------------------------------------------
| Http server configuration
|--------------------------------------------------------------------------
|
| The configuration for the HTTP(s) server. Make sure to go through all
| the config properties to make keep server secure.
|
*/
export const http: ServerConfig = {
  /*
  |--------------------------------------------------------------------------
  | Allow method spoofing
  |--------------------------------------------------------------------------
  |
  | Method spoofing enables defining custom HTTP methods using a query string
  | `_method`. This is usually required when you are making traditional
  | form requests and wants to use HTTP verbs like `PUT`, `DELETE` and
  | so on.
  |
  */
  allowMethodSpoofing: false,

  /*
  |--------------------------------------------------------------------------
  | Subdomain offset
  |--------------------------------------------------------------------------
  */
  subdomainOffset: 2,

  /*
  |--------------------------------------------------------------------------
  | Request Ids
  |--------------------------------------------------------------------------
  |
  | Setting this value to `true` will generate a unique request id for each
  | HTTP request and set it as `x-request-id` header.
  |
  */
  generateRequestId: false,

  /*
  |--------------------------------------------------------------------------
  | Trusting proxy servers
  |--------------------------------------------------------------------------
  |
  | Define the proxy servers that AdonisJs must trust for reading `X-Forwarded`
  | headers.
  |
  */
  trustProxy: proxyAddr.compile('loopback'),

  /*
  |--------------------------------------------------------------------------
  | Generating Etag
  |--------------------------------------------------------------------------
  |
  | Whether or not to generate an etag for every response.
  |
  */
  etag: false,

  /*
  |--------------------------------------------------------------------------
  | JSONP Callback
  |--------------------------------------------------------------------------
  */
  jsonpCallbackName: 'callback',

  /*
  |--------------------------------------------------------------------------
  | Cookie settings
  |--------------------------------------------------------------------------
  */
  cookie: {
    domain: '',
    path: '/',
    maxAge: '2h',
    httpOnly: true,
    secure: false,
    sameSite: false,
  },

  /*
  |--------------------------------------------------------------------------
  | Force Content Negotiation
  |--------------------------------------------------------------------------
  |
  | The internals of the framework relies on the content negotiation to
  | detect the best possible response type for a given HTTP request.
  |
  | However, it is a very common these days that API servers always wants to
  | make response in JSON regardless of the existence of the `Accept` header.
  |
  | By setting `forceContentNegotiationTo = 'application/json'`, you negotiate
  | with the server in advance to always return JSON without relying on the
  | client to set the header explicitly.
  |
  */
  forceContentNegotiationTo: 'application/json',
}

/*
|--------------------------------------------------------------------------
| Logger
|--------------------------------------------------------------------------
*/
export const logger: LoggerConfig = {
  /*
  |--------------------------------------------------------------------------
  | Application name
  |--------------------------------------------------------------------------
  |
  | The name of the application you want to add to the log. It is recommended
  | to always have app name in every log line.
  |
  | The `APP_NAME` environment variable is automatically set by AdonisJS by
  | reading the `name` property from the `package.json` file.
  |
  */
  name: Env.get('APP_NAME'),

  /*
  |--------------------------------------------------------------------------
  | Toggle logger
  |--------------------------------------------------------------------------
  |
  | Enable or disable logger application wide
  |
  */
  enabled: true,

  /*
  |--------------------------------------------------------------------------
  | Logging level
  |--------------------------------------------------------------------------
  |
  | The level from which you want the logger to flush logs. It is recommended
  | to make use of the environment variable, so that you can define log levels
  | at deployment level and not code level.
  |
  */
  level: Env.get('LOG_LEVEL', 'info'),

  /*
  |--------------------------------------------------------------------------
  | Pretty print
  |--------------------------------------------------------------------------
  |
  | It is highly advised NOT to use `prettyPrint` in production, since it
  | can have huge impact on performance.
  |
  */
  prettyPrint: Env.get('NODE_ENV') === 'development',
}

/*
|--------------------------------------------------------------------------
| Profiler
|--------------------------------------------------------------------------
*/
export const profiler: ProfilerConfig = {
  /*
  |--------------------------------------------------------------------------
  | Toggle profiler
  |--------------------------------------------------------------------------
  |
  | Enable or disable profiler
  |
  */
  enabled: true,

  /*
  |--------------------------------------------------------------------------
  | Blacklist actions/row labels
  |--------------------------------------------------------------------------
  |
  | Define an array of actions or row labels that you want to disable from
  | getting profiled.
  |
  */
  blacklist: [],

  /*
  |--------------------------------------------------------------------------
  | Whitelist actions/row labels
  |--------------------------------------------------------------------------
  |
  | Define an array of actions or row labels that you want to whitelist for
  | the profiler. When whitelist is defined, then `blacklist` is ignored.
  |
  */
  whitelist: [],
}

/*
|--------------------------------------------------------------------------
| Validator
|--------------------------------------------------------------------------
|
| Configure the global configuration for the validator. Here's the reference
| to the default config https://git.io/JT0WE
|
*/
export const validator: ValidatorConfig = {}
ks8: apiVersion: v1
kind: Service
metadata:
  name: adonis-kub-projeto
spec:
  type: NodePort
  selector:
      app: adonis-kub-projeto
  ports:
    - port: 8080
      targetPort: 80. apiVersion: apps/v1
kind: Deployment
metadata:
  name: adonis-kub-projeto
  namespace: default
spec:
  replicas: 1
  selector:
    matchLabels:
      app: adonis-kub-projeto
  template:
    metadata:
      labels:
        app: adonis-kub-projeto
    spec:
      containers:
      - name: adonis-kub-projeto
        image: joaoof/adonis-kub-projeto:latest
        ports:
        - containerPort: 80
        env:
          - name: PORT
            value: "3001"
          - name: HOST
            value: "0.0.0.0"
          - name: NODE_ENV
            value: "development"
          - name: APP_KEY
            value: "zmySFzSMOgVl-c91euBqzLjYb0EheK6z"
          - name: DRIVE_DISK
            value: "local"
          - name: CACHE_VIEWS
            value: "false"
          - name: SMTP_HOST
            value: "sandbox.smtp.mailtrap.io"
          - name: SMTP_PORT
            value: "587"
          - name: SMTP_USERNAME
            value: "4cd40d834b414c"
          - name: SMTP_PASSWORD
            value: "a96f084db232e7"
          - name: DB_CONNECTION
            value: "pg"
          - name: PG_HOST
            value: "localhost"
          - name: PG_PORT
            value: "5432"
          - name: PG_USER
            value: "bhayron"
          - name: PG_PASSWORD
            value: "secret123"
          - name: PG_DB_NAME
            value: "sistema"

**Comment by RomainLanz** [comment]
You have to compile your code to use it.

📚 https://docs.adonisjs.com/guides/deployment

**Reply by Joaoof** [reply]
OK, I copied it, the build is done. What's the next step to make this deployment on Kubernetes work?

**Reply by RomainLanz** [reply]
You are using K8s, you must know how it works.

**Reply by Joaoof** [reply]
Yes. I just need to know that if this build is going to go up, or my api in ts? could give me a north.

**Reply by RomainLanz** [reply]
I have not understood your question.

**Reply by Joaoof** [reply]
now when I run ts-node config/app.ts (because before another one gave a syntax error) now it gives an error on line 9. saying that the Env file was not found, or rather, the module. If you want more information, I'll send tsconfig, and package.json

**Reply by Joaoof** [reply]
file in config/app.ts --->
![Captura de tela 2023-08-16 125451](https://github.com/adonisjs/core/assets/113441117/07369130-1551-4aa4-8d81-a57432117e54) quando rodo o comando: 
![Captura de tela 2023-08-16 125559](https://github.com/adonisjs/core/assets/113441117/fae9a30a-ca4c-4d89-8505-b0fd8ec1b903)
package.json: {
  "name": "roleplay-ap",
  "version": "1.0.0",
  "private": true,
  "scripts": {
    "build": "node ace build --production",
    "start": "node ace migration:run && node server.js",
    "dev": "node ace serve --watch",
    "lint": "eslint . --ext=.ts --fix"
  },
  "devDependencies": {
    "@adonisjs/assembler": "^5.9.5",
    "@rocketseat/eslint-config": "^1.2.0",
    "@types/proxy-addr": "^2.0.0",
    "adonis-preset-ts": "^2.1.0",
    "eslint": "^8.44.0",
    "eslint-plugin-adonis": "^2.1.1",
    "pino-pretty": "^10.0.1",
    "typescript": "~4.2",
    "youch": "^3.2.3",
    "youch-terminal": "^2.2.2"
  },
  "dependencies": {
    "@adonisjs/core": "^5.9.0",
    "@adonisjs/env": "^3.0.9",
    "@adonisjs/lucid": "^10.0.0",
    "@adonisjs/repl": "^3.1.11",
    "@adonisjs/view": "^6.2.0",
    "install": "^0.13.0",
    "luxon": "^3.3.0",
    "mysql": "^2.18.1",
    "mysql2": "^3.5.0",
    "proxy-addr": "^2.0.7",
    "reflect-metadata": "^0.1.13",
    "source-map-support": "^0.5.21",
    "swagger-ui": "^5.2.0",
    "swagger-ui-dist": "^5.2.0",
    "swaggify": "^1.0.1",
    "ts-node": "^10.9.1",
    "ts-swagger": "^1.0.0"
  }
}
tsconfig: {
  "extends": "adonis-preset-ts/tsconfig.json",
  "include": [
    "**/*"
  ],
  "exclude": [
    "node_modules",
    "build"
  ],
  "compilerOptions": {
    "outDir": "build",
    "rootDir": "./",
    "sourceMap": true,
    "paths": {
      "App/*": [
        "./app/*"
      ],
      "Config/*": [
        "./config/*"
      ],
      "Contracts/*": [
        "./contracts/*"
      ],
      "Database/*": [
        "./database/*"
      ]
    },
    "types": [
      "@adonisjs/core",
      "@adonisjs/repl",
      "@adonisjs/lucid",
      "@adonisjs/view"
    ]
  }
}

**Comment by Joaoof** [comment]
Error resolvido
